### PR TITLE
Fluid simulation results API

### PR DESCRIPTION
**Pull request by PProfizi**
To do in further PRs:
- Expose a `MeshInfo`
- Use the `MeshInfo` to complete `Zones`
- Use the `Simulation.zones` property to prevent requests on both `FaceZone` and `CellZone`, document it
- Add further examples

### CODE REVIEW

File: src/ansys/dpf/post/simulation.py [hunk 1]
```diff
@@ -102,9 +102,12 @@ def results(self) -> List[str]:
         >>> print(simulation.results) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
         [...]
         """
-        return [
-            str(result) for result in self._model.metadata.result_info.available_results
-        ]
+        return self._model.metadata.result_info.available_results
+
+    @property
```
Comment (cbellot000) [inline src/ansys/dpf/post/simulation.py:107]: @PProfizi could you pease point me to the usage of the mesh info ?

File: src/ansys/dpf/post/zone.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    @property
+    def faces(self) -> List[int]:
+        """Return the list of IDs of the faces in the Zone."""
+        return self._mesh_scoping.ids
+
+
+class Zones:
+    """List of fluid zones."""
+
+    def __init__(self):
```
Comment (cbellot000) [inline src/ansys/dpf/post/zone.py:52]: do we have a way to access zones by name?

Comment (PProfizi) [reply src/ansys/dpf/post/zone.py:52]: @cbellot000 once the MeshInfo is available, sure, but not now

File: src/ansys/dpf/post/simulation.py (hunk 1 above)
Comment (PProfizi) [reply src/ansys/dpf/post/simulation.py:107]: @cbellot000 the MeshInfo is not yet available

File: src/ansys/dpf/post/phase.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ansys.dpf.post.simulation import Simulation
+
+
+class Phase:
+    """Physical state of a given species in a fluid simulation."""
```
[the remark this answers is not in the document]
Comment (PProfizi) [reply src/ansys/dpf/post/phase.py:16]: @cbellot000 I do not think so, no

File: examples/04-Fluid-Examples/00-explore-fluid-simulation.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+print(simulation.result_info["temperature"])
+# Or use an index
+# print(simulation.result_info[12])
+
+###############################################################################
+# Extract the temperature data
+# ----------------------------
+temperature = simulation.temperature()
+print(temperature)
+# # The dataframe obtained shows data for two different phases
```
Comment (cbellot000) [inline examples/04-Fluid-Examples/00-explore-fluid-simulation.py:64]:
```suggestion

# # The dataframe obtained shows data for two different phases
```

File: examples/04-Fluid-Examples/00-explore-fluid-simulation.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+###############################################################################
+# Select data for phase 2 only (Water)
+print(temperature.select(phase=[2]))
+# ###############################################################################
```
Comment (cbellot000) [inline examples/04-Fluid-Examples/00-explore-fluid-simulation.py:68]:
```suggestion

# ###############################################################################
```

File: examples/04-Fluid-Examples/00-explore-fluid-simulation.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+# Or use an index
+# print(simulation.result_info[12])
+
+###############################################################################
+# Extract the temperature data
+# ----------------------------
+temperature = simulation.temperature()
+print(temperature)
+# # The dataframe obtained shows data for two different phases
+###############################################################################
```
Comment (cbellot000) [inline examples/04-Fluid-Examples/00-explore-fluid-simulation.py:65]:
```suggestion
###############################################################################
```
```suggestion

###############################################################################
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -194,18 +195,22 @@ jobs:
     if: startsWith(github.head_ref, 'master') || github.event.action == 'ready_for_review' || !github.event.pull_request.draft
     uses: ./.github/workflows/examples.yml
     with:
-      ANSYS_VERSION: '241'
+      ANSYS_VERSION: "241"
```
Comment (MaxJPRey) [inline .github/workflows/ci.yml:198]: Can you use the environment variable?
```suggestion
      ANSYS_VERSION: ${{env.ANSYS_VERSION}}
```

File: examples/04-Fluid-Examples/00-explore-fluid-simulation.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+# # This argument must be given a list of phase unique identifiers, which appear
+# # in the dataframe in the phase label column between parentheses,
+# # or as listed
+# # under the 'Available qualifier labels' section of the metadata on the result
+# # water_temperature = simulation.temperature(phases=["Copper"])
+# water_temperature = simulation.enthalpy(
+#     phases=[2]
+# )
+# print(water_temperature)
+# # The dataframe obtained now only stores data for the water phase.
```
Comment (MaxJPRey) [inline examples/04-Fluid-Examples/00-explore-fluid-simulation.py:83]: Is it expected to have this entire section commented out.

**Review by rafacanton — APPROVED** [review]

File: examples/04-Fluid-Examples/00-explore-fluid-simulation.py (hunk 8 above)
Comment (PProfizi) [reply examples/04-Fluid-Examples/00-explore-fluid-simulation.py:83]: @MaxJPRey that was to due to a bug server-side which has been resolved in the meantime.
I'll uncomment it.

**Pull request merged**